### PR TITLE
3 story

### PR DESCRIPTION
**Pull request by Leamsi9**

### CODE REVIEW

File: app/cartPage.php [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+<div class="stickyFooterExcluder">
+
+<?php include 'includes/header.php'; ?>
+
+    <div class="container text-center">
+        <h1>Shopping Cart</h1>
+    </div>
+    <div class="container">
+        <div class="row">
+            <table class="table col-xs-6">
```
Comment (ruthwaterfield) [inline app/cartPage.php]: Where are the other 6 columns being used in this row?

File: app/cartPage.php [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                foreach ($bookIds as $bookId) {
+                    $book = new \App\Book($db, $bookId); ?>
+                    <tr>
+                        <td>
+                            <a href='individualBookPage.php?id=<?php echo $book->id; ?>'> <?php echo $book->title; ?></a>
+                        </td>
+                        <td>
+                            <?php echo $book->displayPrice(); ?>
+                        </td>
+                        <td><a class='btn btn-info glyphicon-minus'
```
Comment (ruthwaterfield) [inline app/cartPage.php]: Please put the anchor tag on the line below

File: app/cartPage.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                               href='removeBook.php?id=<?php echo $book->id; ?>&price=<?php echo $book->price; ?>'></a>
```
Comment (ruthwaterfield) [inline app/cartPage.php]: Could you use concatenation  here?

File: app/cartPage.php [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+                        </td>
+                    </tr>
+                <?php } ?>
+            </table>
+        </div>
+    </div>
+    <div class="container">
+        <h5>
```
Comment (ruthwaterfield) [inline app/cartPage.php]: Please put these columns inside a row div

File: app/css/styles.css [hunk 5]
```diff
@@ -89,7 +89,12 @@ a:hover {
     margin: 0;
     line-height: 75px;
 }
-
```
Comment (ruthwaterfield) [inline app/css/styles.css]: please re add this line break

File: app/css/styles.css [hunk 6]
```diff
@@ -89,7 +89,12 @@ a:hover {
...
+.addToCart {
+    height: 79px;
+    font-size: 30px;
+    line-height: 65px;
+    transition: .3s;
+}
 /*footer*/
```
Comment (ruthwaterfield) [inline app/css/styles.css]: Please add a line break before the footer

File: app/individualBookPage.php [hunk 7]
```diff
@@ -47,6 +46,9 @@
             <div class="col-xs-2 col-xs-offset-1 individualBookPrice">
                 <h1><?php echo $individualBook->displayPrice(); ?></h1>
             </div>
+            <a class="col-xs-6 col-xs-offset-1 btn btn-success addToCart" href="addBookToCart.php?id=<?php echo $individualBook->id?>&price=<?php echo $individualBook->price?>">
```
Comment (ruthwaterfield) [inline app/individualBookPage.php]: Can you use concatenation here?

**Review by ruthwaterfield — CHANGES_REQUESTED** [review]

**Review by ruthwaterfield — DISMISSED** [review]

File: app/addBookToCart.php [hunk 8]
```diff
@@ -0,0 +1,12 @@
+<?php
+require "../vendor/autoload.php";
+session_start();
+if (isset($_GET['id']) && isset($_GET['price'])) {
+    $id = $_GET['id'];
+    $price = $_GET['price'];
```
Comment (mporam) [inline app/addBookToCart.php]: technically u dont need this variable

File: app/cartPage.php [hunk 9]
```diff
@@ -0,0 +1,72 @@
+<?php
+require "../vendor/autoload.php";
+session_start();
+$connection = new \App\DbConnector();
+$db = $connection->getDb();
+$bookIds = $_SESSION['cart']['bookIds'];
```
Comment (mporam) [inline app/cartPage.php]: this will give a notice if the session is empty

File: app/cartPage.php [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+        <div class="row">
+            <table class="table">
+                <tr>
+                    <th>Book Title</th>
+                    <th>Price</th>
+                    <th>Remove from cart</th>
+                </tr>
+                <!-- takes the array of book IDs stored in the session and passes them into the Book class to then output title and price into html-->
+                <?php
+                sort($bookIds);
```
Comment (mporam) [inline app/cartPage.php]: this errors if there are no book ids

File: app/cartPage.php [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+                        </td>
+                    </tr>
+                <?php } ?>
+            </table>
+        </div>
+    </div>
+    <div class="container">
+        <div class="row">
+            <h5>
+                <span class="col-xs-2">Books in cart:</span>
```
Comment (mporam) [inline app/cartPage.php]: why is this a span with a col? col classes are floated, which sets display block, so these might as well be divs

File: app/cartPage.php [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+                <span class="col-xs-1"><?php echo (empty($_SESSION)) ? "0" : $_SESSION["cart"]["totalBooks"]; ?></span>
+            </h5>
+            <h5>
+                <span class="col-xs-2">Total price:</span>
```
Comment (mporam) [inline app/cartPage.php]: div

File: app/index.php [hunk 13]
```diff
@@ -14,9 +15,10 @@
     <title>Book List</title>
 </head>
 <body>
-
 <div class="stickyFooterExcluder">
-    <?php include "includes/header.php" ?>
+
+<?php include "includes/header.php" ?>
```
Comment (mporam) [inline app/index.php]: indentation

File: app/removeBook.php [hunk 14]
```diff
@@ -0,0 +1,12 @@
+<?php
+require_once "../vendor/autoload.php";
+session_start();
+
+if(isset($_GET["id"]) && isset($_GET["price"])) {
+    $id = $_GET["id"];
```
Comment (mporam) [inline app/removeBook.php]: again this is redundant

File: app/src/Cart.php [hunk 15]
```diff
@@ -0,0 +1,44 @@
+<?php
+namespace App;
+
+class Cart
+{
+    public static function addBookToCart(array $bookData, int $bookId, float $bookPrice)
```
Comment (mporam) [inline app/src/Cart.php]: where are my doc blocks?!

**Review by mporam — CHANGES_REQUESTED** [review]

**Review by mporam — APPROVED** [review]

**Pull request merged**